enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreCopyClient.java

```diff
@@ -193,28 +200,51 @@ else if ( responseStatus == StoreCopyFinishedResponse.Status.E_STORE_ID_MISMATCH
 
     public static class StoreFileCopyResponseAdaptor extends CatchUpResponseAdaptor<StoreCopyFinishedResponse>
     {
-        private final StoreFileStreams storeFileStreams;
+        private final StoreFileStreamProvider storeFileStreamProvider;
         private final Log log;
-        private String destination;
-        private int requiredAlignment;
+        private StoreFileStream storeFileStream;
 
-        StoreFileCopyResponseAdaptor( StoreFileStreams storeFileStreams, Log log )
+        StoreFileCopyResponseAdaptor( StoreFileStreamProvider storeFileStreamProvider, Log log )
         {
-            this.storeFileStreams = storeFileStreams;
+            this.storeFileStreamProvider = storeFileStreamProvider;
             this.log = log;
         }
 
         @Override
         public void onFileHeader( CompletableFuture<StoreCopyFinishedResponse> requestOutcomeSignal, FileHeader fileHeader )
         {
-            this.destination = fileHeader.fileName();
-            this.requiredAlignment = fileHeader.requiredAlignment();
+            try
+            {
+                storeFileStream = storeFileStreamProvider.acquire( fileHeader.fileName(), fileHeader.requiredAlignment() );
+                requestOutcomeSignal.whenComplete( ( storeCopyFinishedResponse, throwable ) ->
+                {
+                    try
+                    {
+                        storeFileStream.close();
+                    }
+                    catch ( Exception e )
+                    {
+                        log.error( "Unable to close store file stream", e );
+                    }
+                } );
+            }
+            catch ( IOException e )
+            {
+                requestOutcomeSignal.completeExceptionally( e );
+            }
```

Comment:
This looks very similar to what's in `org/neo4j/causalclustering/catchup/storecopy/PrepareStoreCopyResponseAdaptor.java`.

Reply:
Merged them into https://github.com/neo4j/neo4j/pull/11233/commits/fe222fd3cd63919002cd08612196894c6c4cb2c3